R/SEIRD_RU.R

```diff
@@ -0,0 +1,642 @@
+    if (length(trans_params$b) != 1
+        | length(trans_params$k) != 1
+        | length(trans_params$g) != 1
+        | length(trans_params$m) != 1
+        | length(trans_params$C) != 1) {
+      stop("The parameter values should be 1-dimensional.")
+    }
+
+    # check that the connectedness parameter C is between 0 and 1
+    if (trans_params$C < 0 | trans_params$C > 1) {
+      stop("Connectedness parameter C must be in the range 0-1.")
+    }
+    # if all above tests are passed, assign the trans_params namelist to the
+    # object
+    object@transmission_parameters <- trans_params
+
+    return(object)
+  })
+
+#-----------------------------------------------------------------------------
+#' Setter method for contact matrices, or contact rates directly, for urban and
+#' rural communities in the SEIRR_RU model. Matrices must be of type double.
+#' NOTE: if inputting matrices, you MUST assign fraction_rural first, and you 
+#' must input full age demographics, i.e. you cannot only provide the fraction
+#' of the population that is rural!
+#'
+#' @param object (SEIRD_RU model)
+#' @param value (list) list of contact matrices or contact rates.
+#'
+#' @return object of class SEIRD_RU with contact rates assigned.
+#' @export
+
+setGeneric(
+  "contact_rates<-",
+  function(object, value) {
+    standardGeneric("contact_rates<-")
+  })
+
+#' @describeIn SEIRD_RU Setter method for contact matrices for
+#' urban and rural communities in the SEIR model.
+#' Matrices must be of type double.
+#' NOTE: if inputting matrices, you MUST assign fraction_rural first, and you 
+#' must input full age demographics, i.e. you cannot only provide the fraction
+#' of the population that is rural!
+#'
+#' @param object (SEIRD_RU model)
+#' @param value (list) list of contact matrices, with urban first, then rural.
+#'
+#' @return object of class SEIRD_RU with contact matrices
+#' assigned.
+#' @export
+
+setMethod(
+  "contact_rates<-", "SEIRD_RU",
+  function(object, value) {
+
+    # create list of parameter values
+    contact_mat <- value
+
+    # add names to each value
+    names(contact_mat) <- object@contact_rates_names
+
+    # check format of contact matrices
+    if (typeof(contact_mat$urban) != "double"
+        | typeof(contact_mat$rural) != "double") {
+      stop("Contact matrices must be of type double.")
+    }
+    
+    #check if input is matrices or numbers, process accordingly
+    if (length(contact_mat$urban) > 1) {
+      # check if age demographics have been provided
+      if(length(country_demog(object)$urban) == 0 || length(country_demog(object)$rural) == 0) {
+        stop("If you wish to use contact matrices for the model, you must provide
+             vectors of the age demographics of the rural and urban communities
+             BEFORE assigning the contact matrices.")
+      }
+    # compute contacts N_U and N_Y from contact matrices
+    # number of urban contacts for an urban individual, normalized over the
```

Comment:
I think my problem with that is that the populations in the model are all normalized and the units of the entries in the matrix simply aren't "number of contacts" so it seems wrong to note it as such just to simplify things. Since the number of contacts has been divided through by the total population, the numbers in the matrix are much smaller than what you would expect if you were looking at the average number of contacts, so the results we get wouldn't make sense. I understand that I need to phrase it differently cause you don't think it is clear, so I will think about it, but I do think it needs to be noted!

Reply:
I think with respect to your comment above in the vignette it is different because it is a general explanation of the model without reference to our specific implementation of it.